infra/integration-image/Dockerfile.test

```diff
@@ -0,0 +1,37 @@
+# docker build -t vic-test -f Dockerfile.test .
+FROM golang:1.6.2
+
+RUN apt-get update
+RUN apt-get install -y jq time
+RUN curl -sSL https://get.docker.com/ | sh
+
+RUN apt-get install -y gcc python-dev python-setuptools python-pip libffi-dev libssl-dev
+RUN pip install pyasn1 gsutil --upgrade
+RUN pip install robotframework
+RUN pip install robotframework-sshlibrary
+
+# Libraries required to run the docker integration tests
+RUN go get github.com/Sirupsen/logrus
+RUN go get github.com/docker/distribution/digest
+RUN go get github.com/docker/engine-api/types
+RUN go get github.com/docker/engine-api/types/events
+RUN go get github.com/docker/engine-api/types/time
+RUN go get github.com/docker/go-connections/sockets
+RUN go get github.com/go-check/check
+RUN go get github.com/vdemeester/shakers
+RUN go get github.com/docker/libtrust
+RUN go get github.com/gorilla/mux
+RUN go get github.com/docker/libnetwork/driverapi
+RUN go get github.com/godbus/dbus
+RUN go get github.com/kr/pty
+RUN go get github.com/vishvananda/netlink
+RUN go get google.golang.org/grpc
+RUN go get github.com/docker/docker; echo 0
+
+RUN curl -sSL https://github.com/vmware/govmomi/releases/download/v0.7.0/govc_linux_amd64.gz | gzip -d > /usr/local/bin/govc
+RUN chmod +x /usr/local/bin/govc
+
+# Fix because swarm daemon started causing problems for some reason
+RUN rm /go/src/github.com/docker/docker/integration-cli/docker_api_swarm_test.go
+RUN rm /go/src/github.com/docker/docker/integration-cli/daemon_swarm.go
+COPY check_test.go.bak /go/src/github.com/docker/docker/integration-cli/check_test.go
```

Comment:
Assuming we need to have our own check_test.go for some reason (separate doc comment), copying it in as part of the image means we cannot update it and run tests without also rebuilding _and pushing_ the image. That means this image is not a general base for integration testing, it *is* the integration test for a given point in time. Was that the intent?

Is it viable to use something like the following in the drone.yml (not the best way to do this, just a direct translation from Dockerfile to drone.yml):
```
     commands:
      - cp infra/integration-image/check_test.go.bak /go/src/github.com/docker/docker/integration-cli/check_test.go
      - tests/integration-test.sh
```